app/models/singly_linked_list.rb

```diff
@@ -0,0 +1,77 @@
+class SinglyLinkedList
+  include Enumerable
```

Comment:
Enumerable is a very large module.  The reason I avoided including it was that I didn't want to have to assert that things like Enumerable#grep and #include? and #inject? and #map worked.  But, perhaps, they do work if we provide the basic #each?  That'd be very, very slick.  If we don't know that they work, then, I'd suggest not including the module (unless you also provide test-coverage for it).  But, please, what do you think?